### PR TITLE
Sync event timestamp with timegraph

### PR DESCRIPTION
**Pull request by muddana-satish**
Signed-off-by: muddana-satish <satish.muddana@ericsson.com>

### CODE REVIEW

**Comment by bhufmann** [comment]
@muddana-satish I currently reviewing this patch. I'll get back to you soon.

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 1]
```diff
@@ -448,13 +453,43 @@ export class TimeGraphChart extends TimeGraphChartLayer {
...
     protected selectStateInNavigation() {
         const row = this.rowController.selectedRow;
         if (row && this.unitController.selectionRange) {
             const cursorPosition = this.unitController.selectionRange.end;
-            const rowElement = row.states.find((rowElementModel: TimelineChart.TimeGraphState) => rowElementModel.range.start === cursorPosition || rowElementModel.range.end === cursorPosition);
+            const rowElement = row.states.find((rowElementModel: TimelineChart.TimeGraphState) => rowElementModel.range.start === cursorPosition
+                || rowElementModel.range.end === cursorPosition);
             this.selectRowElement(rowElement);
         }
-        this.setNavigationFlag(false);
```
Comment (bhufmann) [inline timeline-chart/src/layer/time-graph-chart.ts:458]: This line was accidentally removed

File: timeline-chart/src/layer/time-graph-chart-cursors.ts [hunk 2]
```diff
@@ -86,7 +86,7 @@ export class TimeGraphChartCursors extends TimeGraphChartLayer {
         this.unitController.onSelectionRangeChange(() => this.update());
     }
 
-    protected maybeCenterCursor() {
+    public maybeCenterCursor() {
```
Comment (bhufmann) [inline timeline-chart/src/layer/time-graph-chart-cursors.ts:89]: Could please separate this into a separate patch, so that we can get the time sync merged first?
Then we can tackle the row synchronization.

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 3]
```diff
@@ -448,13 +453,43 @@ export class TimeGraphChart extends TimeGraphChartLayer {
...
 
+    setEventClickedFlag(flag: boolean) {
+        this.isEventClicked = flag;
+    }
+
+    private selectStateOnEventClicked() {
+        const selectedRow = this.rows.find(row => {
+            if (row && this.unitController.selectionRange) {
+                const cursorPosition = this.unitController.selectionRange.end;
+                const rowElement = row.states.find((rowElementModel: TimelineChart.TimeGraphState) => rowElementModel.range.start === cursorPosition
```
Comment (bhufmann) [inline timeline-chart/src/layer/time-graph-chart.ts:464]: I think we need to find a better way to sync the row than relying the on the timestamp. With nanosecond precision and a huge number of events it will be click on the trace event of start or end state.
I think the row to select needs to determined outside the library and the library needs to provide an API to show (reveal) the selected row.

**Review by bhufmann — CHANGES_REQUESTED** [review]

File: timeline-chart/src/layer/time-graph-chart.ts (hunk 1 above)
Comment (muddana-satish) [reply timeline-chart/src/layer/time-graph-chart.ts]: It's no longer required after #67. But I decided not to modify this file as part of this PR.

**Review by bhufmann — APPROVED** [review]
Looks good to me!

**Review by MatthewKhouzam — APPROVED** [review]
